File: app/src/main/java/com/androidwind/androidquick/sample/mvvm/LiveDataViewModel.java
package com.androidwind.androidquick.sample.mvvm;

import android.app.Application;

import com.androidwind.androidquick.mvvm.livedata.EventMutableLiveData;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

/**
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
public class LiveDataViewModel extends ViewModel {
    private LiveDataRepository repository;
    private EventMutableLiveData<String> mLiveData;

    public EventMutableLiveData<String> getLiveData() {
        return mLiveData;
    }

    public LiveDataViewModel() {
        repository = new LiveDataRepository();
        mLiveData = repository.getLiveData();
    }

    public void test() {
        repository.test();
    }
}

File: sampleapp/src/main/java/com/github/ddnosh/arabbit/sample/jetpack/livedata/LiveDataViewModel.kt
package com.github.ddnosh.arabbit.sample.jetpack.livedata

import androidx.lifecycle.ViewModel
import javax.inject.Inject

/**
 * @author ddnosh
 * @website http://blog.csdn.net/ddnosh
 */
class LiveDataViewModel @Inject constructor() : ViewModel() {
    private val repository: LiveDataRepository = LiveDataRepository()
    val liveData = repository.liveData
    fun test() {
        repository.test()
    }
}
